internal/resources/metal/connection/resource_schema.go

```diff
@@ -82,6 +82,7 @@ func resourceSchema(ctx context.Context) schema.Schema {
 					stringvalidator.OneOf(
 						string(metalv1.INTERCONNECTIONTYPE_DEDICATED),
 						string(metalv1.INTERCONNECTIONTYPE_SHARED),
+						string(metalv1.INTERCONNECTIONTYPE_SHARED_PORT_VLAN),
```

Comment:
The `Description` of this attribute a few lines above should be updated to reflect the new supported value.